### PR TITLE
Leaderboards ProgressBar

### PR DESCRIPTION
**Pull request by mlytvynchuk**

### CODE REVIEW

File: app/src/main/java/com/google/droidjump/LeaderboardsFragment.java [hunk 1]
```diff
@@ -41,42 +41,46 @@
...
     private void fetchLeaderboards() {
-        Games.getLeaderboardsClient(activity,
-                activity.getSavedSignedInAccount())
-                .loadLeaderboardMetadata(/* forceReload= */ false)
-                .continueWithTask(task -> {
-                    LeaderboardBuffer leaderboardBuffer = task.getResult().get();
-                    leaderboards.clear();
+        leaderboards.clear();
+        LoadingHelper.onLoading(activity, getView(), recyclerViewId);
+        Games.getLeaderboardsClient(activity, activity.getSavedSignedInAccount()).loadLeaderboardMetadata(/* forceReload = */ false)
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/LeaderboardsFragment.java:75]: can we get the leaderboards client from activity ?

File: app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java [hunk 2]
```diff
@@ -100,7 +103,15 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        fetchScores(timeSpan, collection);
+    }
+
     private void fetchScores(int timeSpan, int collection) {
+        scores.clear();
+        LoadingHelper.onLoading(activity, getView(), recyclerViewId);
         Games.getLeaderboardsClient(activity, activity.getSavedSignedInAccount())
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java:115]: get the leaderboards client from activity?

File: app/src/main/java/com/google/droidjump/MainActivity.java [hunk 3]
```diff
@@ -61,6 +63,10 @@ public GoogleSignInAccount getSavedSignedInAccount() {
         return savedSignedInAccount;
     }
 
+    public LeaderboardsClient getLeaderboardsClient() {
+        return leaderboardsClient;
+    }
+
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/MainActivity.java:69]: can we move this below the onCreate method?

**Review by redls — APPROVED** [review]

File: app/src/main/java/com/google/droidjump/MainActivity.java (hunk 3 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/MainActivity.java:69]: This one is a public method. I thought that we create our methods in this order:
1. Public
2. Protected
3. Private
Should it be like this?

Comment (redls) [reply app/src/main/java/com/google/droidjump/MainActivity.java:69]: Yes they should. But in this case onCreate seems to be more important than the getter. You can leave it here if you want but you would need to have all the client getters there as well.

Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/MainActivity.java:112]: Fixed.

File: app/src/main/java/com/google/droidjump/MainActivity.java [hunk 4]
```diff
@@ -190,6 +205,7 @@ private void init() {
         setContentView(R.layout.main_activity);
         ((DrawerLayout) findViewById(R.id.drawer_layout))
                 .setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
+        leaderboardsClient = null;
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/MainActivity.java:208]: this should be set to null when the user signs out as well

**Review by redls — APPROVED** [review]

File: app/src/main/java/com/google/droidjump/MainActivity.java [hunk 5]
```diff
@@ -70,6 +64,31 @@ protected void onCreate(Bundle savedInstanceState) {
         init();
```
Comment (mlytvynchuk) [inline app/src/main/java/com/google/droidjump/MainActivity.java:64]: Could I move all fields from init to onCreate?

**Review by redls — APPROVED** [review]

**Pull request merged**